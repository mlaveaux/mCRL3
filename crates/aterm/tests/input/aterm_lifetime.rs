use mcrl3_aterm::ATerm;
use mcrl3_aterm::Symbol;
use mcrl3_aterm::Term;

fn main() {
    let term = {
        let t = ATerm::constant(&Symbol::new("a", 0));
        t.arg(0)
    };
}