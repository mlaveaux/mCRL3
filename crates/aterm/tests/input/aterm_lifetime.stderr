error[E0597]: `t` does not live long enough
 --> tests/input/aterm_lifetime.rs:8:9
  |
6 |     let term = {
  |         ---- borrow later stored here
7 |         let t = ATerm::constant(&Symbol::new("a", 0));
  |             - binding `t` declared here
8 |         t.arg(0)
  |         ^ borrowed value does not live long enough
9 |     };
  |     - `t` dropped here while still borrowed
